internal/rootcoord/ddl_callbacks_create_collection.go

```diff
@@ -216,3 +237,129 @@ func mustConsumeConsistencyLevel(properties []*commonpb.KeyValuePair) (commonpb.
+		}
+
+		logger.Info("creating partition", zap.String("partitionName", partitionName), zap.Int64("partitionID", partID))
+
+		partMsg := message.NewCreatePartitionMessageBuilderV1().
+			WithHeader(&message.CreatePartitionMessageHeader{
+				CollectionId: header.GetCollectionId(),
+				PartitionId:  partID,
+			}).
+			WithBody(&message.CreatePartitionRequest{
+				Base:           commonpbutil.NewMsgBase(commonpbutil.WithMsgType(commonpb.MsgType_CreatePartition)),
+				DbName:         body.DbName,
+				CollectionName: body.CollectionName,
+				PartitionName:  partitionName,
+				DbID:           header.GetDbId(),
+				CollectionID:   header.GetCollectionId(),
+				PartitionID:    partID,
+			}).
+			WithBroadcast(broadcastChannel).
+			MustBuildBroadcast().
+			WithBroadcastID(broadcastID)
+
+		if err := registry.CallMessageAckCallback(ctx, partMsg, result.Results); err != nil {
+			logger.Error("failed to create partition", zap.String("partitionName", partitionName), zap.Error(err))
+			return errors.Wrap(err, "failed to broadcast create partition message")
+		}
+		logger.Info("partition created successfully", zap.String("partitionName", partitionName), zap.Int64("partitionID", partID))
+	}
+
+	// Step 2: Create indexes
+	logger.Info("restoring indexes", zap.Int("indexCount", len(indexInfos)))
+
+	for _, indexInfo := range indexInfos {
+		// Update CollectionID to the new restored collection ID
+		indexInfo.CollectionID = header.GetCollectionId()
+
+		logger.Info("creating index",
+			zap.Int64("indexID", indexInfo.GetIndexID()),
+			zap.String("indexName", indexInfo.GetIndexName()),
+			zap.Int64("fieldID", indexInfo.GetFieldID()))
+
+		createIndexMsg := message.NewCreateIndexMessageBuilderV2().
+			WithHeader(&message.CreateIndexMessageHeader{
+				DbId:         header.GetDbId(),
+				CollectionId: header.GetCollectionId(),
+				FieldId:      indexInfo.GetFieldID(),
+				IndexId:      indexInfo.GetIndexID(),
+				IndexName:    indexInfo.GetIndexName(),
+			}).
+			WithBody(&message.CreateIndexMessageBody{
+				FieldIndex: &indexpb.FieldIndex{
+					IndexInfo:  indexInfo,
+					Deleted:    false,
+					CreateTime: uint64(time.Now().UnixNano()),
+				},
+			}).
+			WithBroadcast([]string{streaming.WAL().ControlChannel()}).
+			MustBuildBroadcast().
+			WithBroadcastID(broadcastID)
+
+		if err := registry.CallMessageAckCallback(ctx, createIndexMsg, result.Results); err != nil {
+			logger.Error("failed to create index",
+				zap.Int64("indexID", indexInfo.GetIndexID()),
+				zap.String("indexName", indexInfo.GetIndexName()),
+				zap.Error(err))
+			return errors.Wrap(err, "failed to broadcast create index message")
+		}
+
+		logger.Info("index created successfully",
+			zap.Int64("indexID", indexInfo.GetIndexID()),
+			zap.String("indexName", indexInfo.GetIndexName()))
+	}
+
+	logger.Info("all indexes restored successfully")
+
+	// Step 3: Restore data
+	restoreResp, err := c.mixCoord.RestoreSnapshotData(ctx, &datapb.RestoreSnapshotRequest{
```

Comment:
it's a little bit weird to put RestoreSnapshotData here. 
move to broadcastRestoreSnapshotV2?